executor/aggfuncs/func_sum.go

```diff
@@ -82,11 +98,77 @@ func (e *sum4Float64) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup [
 			continue
 		}
 		if p.isNull {
-			p.val = input
-			p.isNull = false
+			p.slidingWindowVal.val = input
+			p.slidingWindowVal.isNull = false
+			p.basePartialResult4SumFloat64 = p.slidingWindowVal
+			continue
+		}
+		p.val += input
+		p.slidingWindowVal.val = p.val
+	}
+	return nil
+}
+
+func (e *sum4Float64) Slide(sctx sessionctx.Context, rows []chunk.Row, lastStart, lastEnd uint64, shiftStart, shiftEnd uint64, pr PartialResult) error {
+	p := (*partialResult4SumFloat64)(pr)
+	if shiftStart == 0 && shiftEnd == 0 {
+		if p.isNull {
+			p.basePartialResult4SumFloat64 = p.slidingWindowVal
+		}
+	}
+	for i := uint64(0); i < shiftEnd; i++ {
+		input, isNull, err := e.args[0].EvalReal(sctx, rows[lastEnd+i])
+		if err != nil {
+			return err
+		}
+		if isNull {
 			continue
 		}
+		if p.isNull {
+			if p.slidingWindowVal.isNull {
+				p.slidingWindowVal.val = 0
+				p.slidingWindowVal.isNull = false
+			}
+			p.basePartialResult4SumFloat64 = p.slidingWindowVal
+		}
 		p.val += input
+		p.slidingWindowVal.val = p.val
+	}
+	for i := uint64(0); i < shiftStart; i++ {
+		input, isNull, err := e.args[0].EvalReal(sctx, rows[lastStart+i])
+		if err != nil {
+			return err
+		}
+		if isNull {
+			continue
+		}
+		if p.isNull {
+			if p.slidingWindowVal.isNull {
+				p.slidingWindowVal.val = 0
+				p.slidingWindowVal.isNull = false
+			}
+			p.basePartialResult4SumFloat64 = p.slidingWindowVal
+		}
+		p.val -= input
```

Comment:
Note that is it dangerous to inverse the sum of float types, which may cause loss of precision. MySQL uses `windowing_use_high_precision` to control this kind of optimization, see https://dev.mysql.com/doc/refman/8.0/en/window-function-optimization.html

Reply:
Thinks, I will fix it.